vllm/entrypoints/openai/serving_completion.py

```diff
@@ -150,15 +151,20 @@ async def create_completion(
                     log_tracing_disabled_warning()
 
                 if isinstance(sampling_params, BeamSearchParams):
-                    if not isinstance(self.engine_client, AsyncLLMEngine):
+                    if isinstance(self.engine_client, AsyncLLMEngine):
+                        generator = self.engine_client.beam_search(
+                            prompt_inputs["prompt_token_ids"], request_id_item,
+                            sampling_params)
+                    elif isinstance(self.engine_client, MQLLMEngineClient):
+                        generator = self.engine_client.beam_search(
+                            prompt_inputs["prompt_token_ids"], request_id_item,
+                            sampling_params, lora_request)
+                    else:
                         raise ValueError(
                             "Beam search in the API server is only supported"
-                            " with AsyncLLMEngine. please add "
-                            "`--disable-frontend-multiprocessing` to "
-                            "use beam search.")
-                    generator = self.engine_client.beam_search(
-                        prompt_inputs["prompt_token_ids"], request_id_item,
-                        sampling_params)
+                            " with AsyncLLMEngine and MQLLMEngineClient."
+                            " please add `--disable-frontend-multiprocessing`"
```

Comment:
`--disable-frontend-multiprocessing` will no longer resolve this error. There also is no other case here so this could be an assert.